src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2254,29 +2383,54 @@ static bool isMixableArgument(BoundExpression argument)
                 return true;
             }
 
-            static EscapeArgument getReceiver(MethodSymbol? method, BoundExpression receiver)
+            static EscapeArgument getReceiver(in MethodInfo methodInfo, BoundExpression receiver)
             {
-                if (method is FunctionPointerMethodSymbol)
+                // When there is compound usage the receiver is used once but both the get and 
+                // set methods are invoked. Return the most permissive of the two in order to get
+                // the complete set of ref safety errors.
+                if (methodInfo.Method is not null && methodInfo.SetMethod is not null)
                 {
-                    return new EscapeArgument(parameter: null, receiver, RefKind.None);
+                    var getArgument = getReceiverCore(methodInfo.Method, receiver);
+                    if (getArgument.RefKind == RefKind.Ref)
+                    {
+                        return getArgument;
+                    }
+
+                    var setArgument = getReceiverCore(methodInfo.SetMethod, receiver);
+                    if (setArgument.RefKind == RefKind.Ref)
+                    {
+                        return setArgument;
+                    }
+
+                    return getArgument;
```

Comment:
For sanity I would like to assert that `!getArgument.RefKind.IsWritableReference()` here (it's not an out parameter).